src/Helpers/CreditCardValidator.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+namespace SonarSoftware\CustomerPortalFramework\Helpers;
+
+use Carbon\Carbon;
+
+class CreditCardValidator
+{
+    protected static $cards = [
+        // Debit cards must come first, since they have more specific patterns than their credit-card equivalents.
+
+        CreditCardType::VISAELECTRON => [
+            'type' => CreditCardType::VISAELECTRON,
+            'pattern' => '/^4(026|17500|405|508|844|91[37])/',
+            'length' => [16],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        CreditCardType::MAESTRO => [
+            'type' => CreditCardType::MAESTRO,
+            'pattern' => '/^(5(018|0[23]|[68])|6(39|7))/',
+            'length' => [12, 13, 14, 15, 16, 17, 18, 19],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        CreditCardType::FORBRUGSFORENINGEN => [
+            'type' => CreditCardType::FORBRUGSFORENINGEN,
+            'pattern' => '/^600/',
+            'length' => [16],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        CreditCardType::DANKORT => [
+            'type' => CreditCardType::DANKORT,
+            'pattern' => '/^5019/',
+            'length' => [16],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        // Credit cards
+        CreditCardType::VISA => [
+            'type' => CreditCardType::VISA,
+            'pattern' => '/^4/',
+            'length' => [13, 16, 19],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        CreditCardType::MASTERCARD => [
+            'type' => CreditCardType::MASTERCARD,
+            'pattern' => '/^(5[0-5]|(222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720))/', // 2221-2720, 51-55
+            'length' => [16],
+            'cvcLength' => [3],
+            'luhn' => true,
+        ],
+        CreditCardType::AMEX => [
+            'type' => CreditCardType::AMEX,
+            'pattern' => '/^3[47]/',
+            'format' => '/(\d{1,4})(\d{1,6})?(\d{1,5})?/',
```

Comment:
Nit: This appears to be a carry over that is not used / needed.  Would remove for greater consistency among the cards.